ably/ably_test.go

```diff
@@ -112,3 +113,16 @@ func (r connMock) Receive(deadline time.Time) (*proto.ProtocolMessage, error) {
 func (r connMock) Close() error {
 	return r.CloseFunc()
 }
+func assertNil(t *testing.T, object interface{}) {
+	value := reflect.ValueOf(object)
```

Comment:
This should first check that `object != nil`, [or this will panic sometimes](https://play.golang.org/p/yhxkgUzO1Ou). ([This is why](https://www.calhoun.io/when-nil-isnt-equal-to-nil/).)